assets/js/components/notifications/BannerNotification/index.js

```diff
@@ -165,7 +164,9 @@ export default function BannerNotification( props ) {
 	};
 
 	const dismissNotification = () => {
-		setIsClosed( true );
+		if ( isMounted() ) {
```

Comment:
This prevents the "Can't perform a React state update on an unmounted component" error we've been seeing with this component.